ethereum/balancer/view_trades.sql

```diff
@@ -139,21 +139,21 @@ WITH rows AS (
             t."tokenOut" AS token_a_address,
             t."tokenIn" AS token_b_address,
             t."poolId" AS exchange_contract_address,
-            "swapFeePercentage"/1e18 AS swap_fee,
+            COALESCE(s1."swapFeePercentage", s2."swapFeePercentage")/1e18 AS swap_fee,
             t.evt_tx_hash AS tx_hash,
             NULL::integer[] AS trace_address,
             t.evt_index
         FROM
             balancer_v2."Vault_evt_Swap" t
-        LEFT JOIN balancer_v2."WeightedPool_evt_SwapFeePercentageChanged" s ON s.contract_address = SUBSTRING(t.exchange_contract_address from 0 for 21)
-        AND s.evt_block_time = (
+        LEFT JOIN balancer_v2."WeightedPool_evt_SwapFeePercentageChanged" s1 ON s1.contract_address = SUBSTRING(t."poolId" from 0 for 21)
+        AND s1.evt_block_time = (
             SELECT MAX(evt_block_time)
             FROM balancer_v2."WeightedPool_evt_SwapFeePercentageChanged"
             WHERE evt_block_time <= t.evt_block_time
             AND contract_address = SUBSTRING(t."poolId" from 0 for 21)
         )
-        LEFT JOIN balancer_v2."WeightedPool2Tokens_evt_SwapFeePercentageChanged" s ON s.contract_address = SUBSTRING(t."poolId" from 0 for 21)
-        AND s.evt_block_time = (
+        LEFT JOIN balancer_v2."WeightedPool2Tokens_evt_SwapFeePercentageChanged" s2 ON s2.contract_address = SUBSTRING(t."poolId" from 0 for 21)
```

Comment:
Any chance you can rewrite this one to merge `WeightedPool2Tokens...` and `WeightedPool..`? 
Can't distinguish between those 2 on decoding side, atm. :\ Had to merge v1 and v2 pools under `WeightedPool`.